spec/models/login_spec.rb

```diff
@@ -23,6 +23,30 @@
     expect(login).to be_valid
   end
 
+  it 'validates uniqueness of identification' do
+    login = described_class.new(identification: 'test@example.com', oauth2_token: 'token', uid: '123', provider: 'facebook')
+    login.save
+    login2 = described_class.new(identification: 'test@example.com', oauth2_token: 'token2', uid: '123', provider: 'facebook')
+
+    expect(login2).to_not be_valid
+  end
+
+  it 'validates uniqueness of oauth2_token' do
+    login = described_class.new(identification: 'test@example.com', oauth2_token: 'token', uid: '123', provider: 'facebook')
+    login.save
+    login2 = described_class.new(identification: 'test2@example.com', oauth2_token: 'token', uid: '123', provider: 'facebook')
+
+    expect(login2).to_not be_valid
+  end
+
+  it 'validates uniqueness of single_use_oauth2_token' do
+    login = described_class.new(single_use_oauth2_token: 'oldtoken')
+    login.save
+    login2 = described_class.new(single_use_oauth2_token: 'oldtoken')
+
+    expect(login2).to_not be_valid
+  end
```

Comment:
You can use the `validate_uniqueness_of` matcher here: http://www.rubydoc.info/gems/shoulda-matchers/Shoulda%2FMatchers%2FActiveRecord%3Avalidate_uniqueness_of